src/librustc/middle/traits/object_safety.rs

```diff
@@ -104,9 +110,6 @@ fn object_safety_violations_for_trait<'tcx>(tcx: &ty::ctxt<'tcx>,
         .collect();
 
     // Check the trait itself.
-    if trait_has_sized_self(tcx, trait_def_id) {
```

Comment:
Could this perhaps be left in the same place but check that the violations array doesn't already contain `SizedSelf`?